doc/TOR.md

```diff
@@ -100,6 +185,51 @@ is necessary to set up a [Tor Hidden Service].
 *NOTE: In the case where only non-persistent addresses are required,
 you don't have to create the hidden service and you can skip this part.*
 
+##### Automatic persistent .onion address
+
+It is possible to generate persistent .onion addresses automatically.
+
+Add the following lines in the `/etc/tor/torrc` file
+(you might already have done this if for example you connected Bitcoin
+over Tor):
+
+````
+ControlPort 9051
+CookieAuthentication 1
+CookieAuthFileGroupReadable 1
+````
+
+Then you can use `--addr=statictor:127.0.0.1:9051` instead of
+`--announce-addr=.onionAddressV3`.
+By default V3 onion addresses are generated.
+
+Note that you have to specify a `--bind-addr` first before using
+`--addr=statictor:`.
+Generally `--bind-addr=127.0.0.1:9735` should work fine.
+
+You can also have multiple persistent .onion addresses
+by adding `/torblob=BLOB`, where `BLOB` is 32 to 64 ***random***
+bytes of text.
+Note that this blob will be used to derive the secret key behind
+the .onion address and you should keep the blob secret otherwise
+anyone who steals it can spoof your node.
```

Comment:
```suggestion
anyone who steals it can spoof your address.
```
The node identity is still bound to the `hsm_secret` not the onion blob. A squatter can only self-assign the onion address preventing you from getting incoming connections through that address.

Reply:
In general one could say this blob cmd is not intended and designed to be worry free, 
Its a mighty cmd and **the blob is the private key of this tor onion address**.
If the user is not careful in the randomness and lose that information like you say, that inbound tor connection could easly be blocked or MITM.   
Its use is quite questionable and maybe we should change or remove it? , but it is i,e,an easy way to get inbound reached behind a nat or fw when ppl only know say the id  or name of the node and have no address and the user defined i.e. the first 16 hex from the node id as blob and the rest 32 char with some randomness.
If the node id is known to the public this allows like you point out quite easy blocking, if not, its a quite private way to get first hidden inbound and then escalate hiddeness or services i,e, signing or new address on the fly by paying an invoice that triggers a script, since that traffic is encoded as you point out by our hsm we should be insofar on every step quite safe.

so maybe
> anyone who know that secret can spoof your address and block income traffic from that address

Reply:
I might have formulated this wrongly, I don't think that generating the blob from which we generate the onion key-pair shouldn't be securely generated, I'm just saying that the way it is formulated now (spoof your node) is not correct, since the Noise_XK handshake authenticates both endpoints. The risk of the tor blob leaking is limited to address spoofing, while the node cannot be impersonated and the funds are not at risk :wink:

Reply:
Yes correct, i also feel strong safe on the c/ln side even if that blob done wrong might dox us or leak on and in tor, so your point is absolutely correct the node is not touched by what the transport does, that is a strong side of ln we should promote to users at every level.
And that node wording should be changed.
siggh documentation is not my turf  ... :smile:

Reply:
if PR #3439 gets merged, even pealing the Tor onion should be no longer possible if blob data say less than 48 char get lost and if greater 16 collisions are not likely.